languageserver/src/main/scala/langserver/utils/JsonRpcUtils.scala

```diff
@@ -34,8 +34,8 @@ object JsonRpcUtils {
    *         )
    * }}}
    */
-   def valueFormat[A, B: Format](apply: B => A)(unapply: A => B): Format[A] = new Format[A] {
+   def valueFormat[A, B: Format](apply: B => A)(unapply: A => B): OFormat[A] = new OFormat[A] {
     override def reads(json: JsValue) = Reads.of[B].reads(json).map(apply(_))
-    override def writes(o: A) = Writes.of[B].writes(unapply(o))
+    override def writes(o: A) = Writes.of[B].writes(unapply(o)).as[JsObject]
```

Comment:
This seems to be the cause of the exception I reported earlier. In the case of `DocumentSymbolResult` the underlying type is a sequence, not a Json object.

Reply:
If that's the case we'll need a similar workaround like the one I implemented for the Locations for "textDocument/definition". It seems that the latest play-json-rcp Message.MessageFormats only accepts JsObject, can't deal with JsValue. I'll take a look later today.
Can you add a test case to include this case?

Reply:
I pushed the changes to fix the issue with "textDocument/definition". I had to use the same workaround used for the Seq[Location]. Changed JsonRpcUtils.valueFormat to work only with JsObject.